students/erica_edwards/Lesson03/mailroom.py

```diff
@@ -0,0 +1,97 @@
+
+from operator import itemgetter
+
+Donors = [('Sandy Pie', [75]),
+          ('Judy Smith', [75, 100, 1000]),
+          ('Mike Jones', [75, 1000]),
+          ('Joe Smith', [75, 100, 2000]),
+          ('Kelly Blue', [75, 150, 275])]
+
+def menu():
+    quit = False
+    while not quit:
+        print()
+        print('''Menu:   
+                    1. Send a Thank You.
+                
+                    2. Create a Report.
+            
+                    3. Quit.''')
+        print()
+        response = input("Please enter a number to make your selection. ")
+        if response == '1':
+            donor_name()
+            
+        if response == '2':
+            totals()    
+
+        if response == '3':
+            quit = True
+
+def list_donors():
+    print(Donors)
+    donor_name()
+
+def donor_name():
+    done = False
+    
+    # While collecting user input.
+    while not done:
+        donorName = input("Enter the full name of the donor. (or q to quit) ")
+        
+        if donorName == 'q':
+            return
+        # If user types list print Donors       
+        elif donorName.upper() == "LIST":
+            list_donors()
+            continue
+        else:  # user didn't type list    
+            if donorName not in [item[0] for item in Donors]:
```

Comment:
it would be more efficient to use `for` loop and exit out of the loop as soon as donor is found.

Reply:
Is this what you mean? 
```python
            for item in donors:
                if item[0] == donor_name:
                    break
                else: 
                    donors.append((donor_name, list()))
```

Reply:
close 🙂 

```
for donor in donors:
    if donor[0] == donor_name:
        donor[1].append(donation)  # update existing donor
else:  # notice indentation here
    # new donor
    donors.append((name, [donation]))
```